src/reynir_correct/pattern.py

```diff
@@ -1771,6 +2270,94 @@ def wrong_noun(nouns: Set[str], tree: SimpleTree) -> bool:
             )
         )
 
+        # 'af' incorrectly used with particular nouns
+        def wrong_noun_af(nouns: Set[str], tree: SimpleTree) -> bool:
+            """ Context matching function for the %noun macro in combination
+                with 'af' """
+            lemma = tree.own_lemma
+            if not lemma:
+                # The passed-in tree node is probably not a terminal
+                return False
+            try:
+                case = (set(tree.variants) & {"nf", "þf", "þgf", "ef"}).pop()
+            except KeyError:
+                return False
+            return (lemma + "_" + case) in nouns
+
+        NOUNS_AF: Set[str] = {
```

Comment:
FrozenSet and frozenset((tuple, ...)) would be slightly better here and more efficient